todo_app/app.py

```diff
@@ -1,41 +1,44 @@
-from flask import Flask, render_template, request, redirect, url_for
-from todo_app.trello_api import TrelloApi
+from flask import Flask, render_template, redirect, request, url_for
+from todo_app.mongo_db import MongoDb
 from todo_app.card_services import CardService
 from todo_app.view_model import ViewModel
+from todo_app.card import Card
 
 
-def create_app():
+def create_app():    
     app = Flask(__name__)    
-    trello_api = TrelloApi()
+    mongo_db = MongoDb()
     card_service = CardService()
+    collection = mongo_db.get_mongo_db_collection()
 
     @app.route('/')
-    def index():
-        trello_cards = card_service.card_details(trello_api.get_board_cards())
-        item_view_model = ViewModel(trello_cards)    
+    def index():        
+        items = MongoDb.get_items(collection)        
+        item_view_model = ViewModel(items)    
         return render_template('index.html', view_model=item_view_model)
 
     @app.route('/', methods=["POST"])
     def add_item():
         new_todo_item = request.form["NewToDoItem"]
-        trello_api.add_card(new_todo_item)
+        mongo_db.add_card(new_todo_item)
         return redirect(url_for('index'))
 
     @app.route('/items/<id>/inprogress')
     def move_item_to_in_progress(id):
-        trello_api.move_card_not_started_in_progress(id)
+        mongo_db.move_card_not_started_in_progress(id)
         return redirect(url_for('index'))
 
     @app.route('/items/<id>/done')
     def move_item_to_done(id):
-        trello_api.move_card_in_progress_done(id)
+        mongo_db.move_card_in_progress_done(id)
         return redirect(url_for('index'))
 
     @app.route('/items/<id>/todo')
     def move_item_todo(id):
-        trello_api.move_card_done_todo(id)
+        mongo_db.move_card_done_todo(id)
         return redirect(url_for('index'))
 
     return app
 
+
 app = create_app()
```

Comment:
At the moment `poetry run pytest` from a terminal outside of VSCode fails because this tries to run `create_app()` at import time, which is before your tests have had a chance to load environment variables.

Since line 12 `collection = mongo_db.get_mongo_db_collection()` tries to connect to Mongo this fails.

This change means it will only run `create_app()` if `app.py` is the "main" file, i.e. the file being run. You can probably also get rid of this line entirely, but check your app still runs if you do!

```suggestion
if __name__ == '__main__':
    app = create_app()
```